brave/brave-ios
# Fix #5577: Solana/Filecoin network switch

## Summary of Changes
- Add support for switching to Solana and Filecoin networks

This pull request fixes #5577 

## Submitter Checklist:


## Test Plan:
0. Until Solana support work is ready, need to add `.sol` and/or `.fil` to `supportedCoinTypes` in `WalletConstants.swift`.
1. Switch network to a Solana network.
2. Verify prompted via alert to create a Solana account.
3. Tap 'No'
4. Verify network remains on previous Ethereum network.
5. Switch network to a Solana network
6. Tap 'Yes' on create account alert
7. Verify Add Account modal shows up, and does not show the coin selection view (we know coin type from selected network).
    -  NOTE: Actually creating a Solana / Filecoin account is not supported yet. Add account dismissal will check for accounts > 0 for that coin type. If none exist, you will remain on previous network.
8. Dismiss Add Account modal
9. Verify network remains on previous Ethereum network.


## Screenshots:

https://user-images.githubusercontent.com/5314553/179073380-9feb80c8-3b64-467c-8285-25aefaffa026.mp4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
